internal/controller/nonadminbackupstoragelocation_controller.go

```diff
@@ -377,6 +413,180 @@ func (r *NonAdminBackupStorageLocationReconciler) setFinalizerOnNaBSL(ctx contex
+
+	if !reflect.DeepEqual(nabslRequest.Status.VeleroBackupStorageLocationRequest.DeepCopy().RequestedSpec, nabsl.Spec.BackupStorageLocationSpec) {
+		message = "NaBSL Spec update not allowed. Changes will not be applied. Delete NaBSL and create new one with updated spec"
+		updatedRejectedCondition = meta.SetStatusCondition(&nabsl.Status.Conditions, metav1.Condition{
+			Type:    string(nacv1alpha1.NonAdminBSLConditionSpecUpdateApproved),
+			Status:  metav1.ConditionFalse,
+			Reason:  "BslSpecUpdateRejected",
+			Message: message,
+		})
+		statusCondition = metav1.ConditionTrue
+		expectedPhase = nabsl.Status.Phase
+		terminalErr = reconcile.TerminalError(errors.New(message))
+	} else if nabslRequest.Status.VeleroBackupStorageLocationRequest.NACUUID == constant.EmptyString || nabslRequest.Status.VeleroBackupStorageLocationRequest.NACUUID != nabsl.Status.VeleroBackupStorageLocation.NACUUID {
+		message = "NonAdminBackupStorageLocationRequest does not contain valid NAC UUID and can not be approved"
+		updatedRejectedCondition = meta.SetStatusCondition(&nabsl.Status.Conditions, metav1.Condition{
+			Type:    string(nacv1alpha1.NonAdminBSLConditionApproved),
+			Status:  metav1.ConditionFalse,
+			Reason:  "BslSpecUpdateRejected",
+			Message: message,
+		})
+		terminalErr = reconcile.TerminalError(errors.New(message))
+		expectedPhase = nacv1alpha1.NonAdminPhaseBackingOff
+	} else {
+		switch {
+		case nabslRequest == nil:
+			shouldRequeue = true
+			reason, message = "BslSpecApprovalPending", "NonAdminBackupStorageLocationRequest not found, creating one"
+		case nabslRequest.Spec.ApprovalDecision == "pending" || nabslRequest.Spec.ApprovalDecision == constant.EmptyString:
+			reason, message = "BslSpecApprovalPending", "NonAdminBackupStorageLocationRequest approval pending"
+			terminalErr = reconcile.TerminalError(errors.New(message))
+		case nabslRequest.Spec.ApprovalDecision == "approve":
+			statusCondition = metav1.ConditionTrue
+			reason, message = "BslSpecApproved", "NonAdminBackupStorageLocationRequest approval decision set to Approve"
+		case nabslRequest.Spec.ApprovalDecision == "reject":
+			reason, message = "BslSpecRejected", "NonAdminBackupStorageLocationRequest approval decision set to Reject"
+			expectedPhase = nacv1alpha1.NonAdminPhaseBackingOff
+			terminalErr = reconcile.TerminalError(errors.New(message))
+		default:
+			reason, message = "BslSpecInvalid", "NonAdminBackupStorageLocationRequest approval decision is invalid"
+			expectedPhase = nacv1alpha1.NonAdminPhaseBackingOff
+			terminalErr = reconcile.TerminalError(errors.New(message))
+		}
+		updatedApprovedCondition = meta.SetStatusCondition(&nabsl.Status.Conditions, metav1.Condition{
+			Type:    string(nacv1alpha1.NonAdminBSLConditionApproved),
+			Status:  statusCondition,
+			Reason:  reason,
+			Message: message,
+		})
+	}
+
+	updatePhase := updateNonAdminPhase(&nabsl.Status.Phase, expectedPhase)
+
+	if statusCondition == metav1.ConditionFalse {
```

Comment:
Should we also check the type of condition here ?

Reply:
No need as we have full control over this variable and we set this variable on line:
https://github.com/migtools/oadp-non-admin/pull/216/files/30d565184e6f17c9f1a64af684ed86f2eed90018#diff-ab1df5cab3611d202a4b22ab31a471f15bd9b010c803a8df03b55ce6b7e09c2eR433

So it's not value which we get from the cluster from the condition, but we know for which condition we are setting it to true/false.

I've renamed this var to be more informative: `adminApprovedCondition`